rebasehelper/tags.py

```diff
@@ -82,23 +84,43 @@ def _map_sections_to_parsed(cls, raw_content: SpecContent, parsed_content: SpecC
 
     @classmethod
     def _parse(cls, raw_content: SpecContent, parsed_content: SpecContent) -> Tuple[Tag]:
-        result = []
+        result: List[Tag] = []
         parsed_mapping = cls._map_sections_to_parsed(raw_content, parsed_content)
+        patch_source_indices = {tag: 0 for tag in ('source', 'patch')}
```

Comment:
I don't like using dict for this. I'd rather see two (function?) attributes, called `next_source_index` and `next_patch_index` for example. Not to mention `indices` is inconsistent with the rest of this module.

Reply:
By using dict we can easily modify the indices inside the methods (for example in `_sanitize`) and the changes "are propagated" to the calling method (`Tags._parse`). That wouldn't be the case with just integers, would it? Perhaps the 2 attributes you mentioned could be attributes of `Tags` but that doesn't seem right to me

Reply:
Yes, I know, that's what I don't like about it. Why not? Also, I suggested function attributes, because their scope would be exactly just `Tags._parse()`.